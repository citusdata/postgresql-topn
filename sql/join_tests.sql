-- Onder dreamed about these queries crashing
SELECT
    (topn(topn_add_agg(customer_id), 10)).*, (topn(topn_add_agg(product_id), 10)).*
FROM
    customer_reviews
ORDER BY
    2 DESC, 4 DESC, 1, 3;

SELECT
    *
FROM
    (SELECT
        product_category, (topn(topn_add_agg(customer_id), 10)).*
    FROM
        customer_reviews
    GROUP BY
        1
    ) customers
JOIN
    (SELECT
        product_category, (topn(topn_add_agg(product_id), 10)).*
    FROM
        customer_reviews
    GROUP BY
        1
    ) products
ON
    (customers.product_category = products.product_category)
ORDER BY
    3 DESC,6 DESC, 1, 2, 4, 5
LIMIT
    20;
