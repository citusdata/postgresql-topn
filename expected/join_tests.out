-- Onder dreamed about these queries crashing
SELECT
    (topn(topn_add_agg(customer_id), 10)).*, (topn(topn_add_agg(product_id), 10)).*
FROM
    customer_reviews
ORDER BY
    2 DESC, 4 DESC, 1, 3;
      item      | frequency |    item    | frequency 
----------------+-----------+------------+-----------
 ATVPDKIKX0DER  |    331760 | 0743527550 |       576
 AGHIV0V0ON7MO  |       412 | 0613033205 |       576
 ACCH00DRABL57  |       210 | 0671014730 |       576
 A1EDOWER7RO6KZ |       184 | 0671003755 |       576
 A3HF6NQ541X7HW |       181 | 0871136791 |       560
 ALU2IJ2DZD86X  |       177 | 0679460691 |       560
 A84J7PUWOC138  |       172 | 0375700757 |       560
 A2FAGKHE7DRFXA |       151 | 0375402926 |       542
 A2JS2OU5SBH1XQ |       149 | 0613221583 |       531
 A3AZ4O4I9S4668 |       146 | 1590073991 |       531
(10 rows)

SELECT
    *
FROM
    (SELECT
        product_category, (topn(topn_add_agg(customer_id), 10)).*
    FROM
        customer_reviews
    GROUP BY
        1
    ) customers
JOIN
    (SELECT
        product_category, (topn(topn_add_agg(product_id), 10)).*
    FROM
        customer_reviews
    GROUP BY
        1
    ) products
ON
    (customers.product_category = products.product_category)
ORDER BY
    3 DESC,6 DESC, 1, 2, 4, 5
LIMIT
    20;
   product_category   |     item      | frequency |   product_category   |    item    | frequency 
----------------------+---------------+-----------+----------------------+------------+-----------
 Literature & Fiction | ATVPDKIKX0DER |     56898 | Literature & Fiction | 0613033205 |       578
 Literature & Fiction | ATVPDKIKX0DER |     56898 | Literature & Fiction | 0671003755 |       578
 Literature & Fiction | ATVPDKIKX0DER |     56898 | Literature & Fiction | 0671014730 |       578
 Literature & Fiction | ATVPDKIKX0DER |     56898 | Literature & Fiction | 0375700757 |       560
 Literature & Fiction | ATVPDKIKX0DER |     56898 | Literature & Fiction | 0679460691 |       560
 Literature & Fiction | ATVPDKIKX0DER |     56898 | Literature & Fiction | 0871136791 |       560
 Literature & Fiction | ATVPDKIKX0DER |     56898 | Literature & Fiction | 0613222407 |       526
 Literature & Fiction | ATVPDKIKX0DER |     56898 | Literature & Fiction | 0446605980 |       418
 Literature & Fiction | ATVPDKIKX0DER |     56898 | Literature & Fiction | 0385315236 |       400
 Literature & Fiction | ATVPDKIKX0DER |     56898 | Literature & Fiction | 0440222656 |       400
 Children's Books     | ATVPDKIKX0DER |     37233 | Children's Books     | 0881030392 |       220
 Children's Books     | ATVPDKIKX0DER |     37233 | Children's Books     | 0345413350 |       217
 Children's Books     | ATVPDKIKX0DER |     37233 | Children's Books     | 037582345X |       217
 Children's Books     | ATVPDKIKX0DER |     37233 | Children's Books     | 0440418321 |       217
 Children's Books     | ATVPDKIKX0DER |     37233 | Children's Books     | 0679879242 |       217
 Children's Books     | ATVPDKIKX0DER |     37233 | Children's Books     | 0679893105 |       217
 Children's Books     | ATVPDKIKX0DER |     37233 | Children's Books     | 0060281375 |       153
 Children's Books     | ATVPDKIKX0DER |     37233 | Children's Books     | 0064405370 |       153
 Children's Books     | ATVPDKIKX0DER |     37233 | Children's Books     | 0064409392 |       153
 Children's Books     | ATVPDKIKX0DER |     37233 | Children's Books     | 0066238501 |       153
(20 rows)

